src/utils/detectors.js

```diff
@@ -17,7 +17,11 @@ export const importLocalName = (name, state, bypassCache = false) => {
     return localNameCache[cacheKey]
   }
 
-  let localName = name === 'default' ? 'styled' : name
+  let localName = state.styledRequired
+    ? name === 'default'
+      ? 'styled'
+      : name
+    : false
```

Comment:
For the ESM case we want to start with the assumption there is no import and work from there, rather than assume the `styled` import is present and named in that precise way. In the case of the `css` prop we then will dynamically-inject the import as needed.